src/ngraph/runtime/allocator.hpp

```diff
@@ -0,0 +1,50 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include "ngraph/except.hpp"
+#include "ngraph/util.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        class Allocator;
+        class DefaultAllocator;
+        /// \brief Returns a statically allocated default ngraph allocator
+        //         that calls into system allocation libraries
+        Allocator* get_default_allocator();
```

Comment:
Again - as I mentioned, please either return a unique_ptr OR return a raw pointer from the allocator stored as a member variable. Best is to create a unique_ptr and return the raw pointer from it.